framework/scripts/generate_test_list.py

```diff
@@ -55,45 +59,50 @@ def main():
 
     # read and process all the arguments
     for name, file in test_list_files.items():
-        with open(file, "r") as f:
+        with open(file, 'r') as f:
             lines = f.readlines()
 
         test_lists[name] = []
         test_list = test_lists[name]
         for l in lines:
-            l = l.split("#", 1)[0]
+            l = l.split('#', 1)[0]
             l = l.strip()
             if len(l) == 0:
                 continue
             test_list.append(l)
             all_tests[l] = 1
 
     # generate the .cpp file
-    with open(cpp_file, "w") as f:
+    with open(cpp_file, 'w') as f:
+        f.write('#include <span>\n')
         f.write(f'#include "{h_file_basename}"\n\n')
-        decls = [ f"extern struct test _test_{test};" for test in sorted(all_tests.keys()) ]
-        f.write("\n".join(decls))
-        f.write("\n\n")
+        f.write('using namespace std;\n\n')
+        decls = [ f'extern struct test _test_{test};' for test in sorted(all_tests.keys()) ]
+        f.write('\n'.join(decls))
+        f.write('\n\n')
         for name, tests in test_lists.items():
-            f.write(f"static constexpr struct test *{name}_test_list[] = {{")
-            test_list = [ f"    &_test_{test}" for test in tests ]
-            f.write(",\n".join(test_list))
-            f.write("\n};\n\n")
+            f.write(f'static constexpr struct test * const {name}_test_list_[] = {{')
+            test_list = [ f'    &_test_{test}' for test in tests ]
+            f.write(',\n'.join(test_list))
+            f.write('\n};\n\n')
+            f.write(f'static constinit const span<struct test * const> {name}_test_list({name}_test_list_);\n\n');
         # write selector function
-        f.write("struct test * const *get_test_list(const char *platform, size_t *size) {\n")
+        f.write('optional<const span<struct test * const>> get_test_list(const char *test_list_name) {\n')
         if default_name is not None:
-            f.write(f'    if (!platform) {{ *size = sizeof({default_name}_test_list)/sizeof(struct test *); return (struct test * const *)&{default_name}_test_list; }}\n')
+            f.write(f'    if (!test_list_name) return {default_name}_test_list;\n')
         for name in test_lists.keys():
-            f.write(f'    if  (!strcmp(platform, "{name}")) {{ *size = sizeof({name}_test_list)/sizeof(struct test *); return (struct test * const *)&{name}_test_list; }}\n')
-        f.write("    return nullptr;\n")
-        f.write("}\n")
+            f.write(f'    if  (!strcmp(test_list_name, "{name}")) return {name}_test_list;\n')
```

Comment:
Minor coding style suggestion: don't declare the spans as `static constinit` variables. Just return the actual array directly in the `return` statement.